uber/site_sections/registration.py

```diff
@@ -801,6 +801,54 @@ def review(self, session):
                                     .order_by(Attendee.full_name).all()}
 
     @site_mappable
+    def promo(self, session, message='', **params):
+        pc = session.promo_code(params)
+        now = datetime.now(UTC)
+        today = "%d-%02d-%02d" % (now.date().year, now.date().month, now.date().day)
+        max_day = "%d-%02d-%02d" % (c.EPOCH.date().year, c.EPOCH.date().month, c.EPOCH.date().day)
+        if 'price' in params:
+            if pc.expiration_date == '':
```

Comment:
This can be rewritten as:
```
if pc.expiration_date:
  year, month, day = pc.expiration_date.split("-")
  pc.expiration_date = datetime(int(year), int(month), int(day)).replace(tzinfo=UTC)
```

Python evaluates empty strings to falsey, so you usually don't have to explicitly check for `''`.

That being said, I'm not super sure why this if block is required. SQLAlchemy should be handling the datetime stuff for you. Was there an error?